bentoml/cli/bento.py

```diff
@@ -218,3 +220,43 @@ def delete(bentos, yes):
                     CLI_COLOR_ERROR,
                 )
             _echo(f'BentoService {name}:{version} deleted')
+
+    @cli.command(
+        help='Retrieves BentoService artifacts into a target directory',
+        short_help="Retrieves BentoService artifacts into a target directory",
+    )
+    @click.argument("bento", type=click.STRING)
+    @click.option(
+        '--target_dir',
+        help="Directory to put artifacts into. Defaults to pwd.",
+        default=os.getcwd(),
+    )
+    def retrieve(bento, target_dir):
+        if ':' not in bento:
+            _echo(f'BentoService {bento} invalid - specify name:version')
+            return
+        name, version = bento.split(':')
+
+        yatai_client = YataiClient()
+
+        track_cli('bento-retrieve')
+        get_bento_result = yatai_client.repository.get(name, version)
+        if get_bento_result.status.status_code != status_pb2.Status.OK:
+            error_code, error_message = status_pb_to_error_code_and_message(
+                get_bento_result.status
+            )
+            _echo(
+                f'BentoService {name}:{version} not found - '
+                f'{error_code}:{error_message}',
+                CLI_COLOR_ERROR
+            )
+            return
+
+        if get_bento_result.bento.uri.s3_presigned_url:
+            bento_service_bundle_path = get_bento_result.bento.uri.s3_presigned_url
+        else:
+            bento_service_bundle_path =  get_bento_result.bento.uri.uri
+
+        bento_path = safe_retrieve(bento_service_bundle_path, target_dir)
+
+        click.echo('Service %s artifact directory => %s' % (name, bento_path))
```

Comment:
should this be  ``` click.echo('Service %s artifact directory => %s' % (name, target_dir))``` instead?  if safe_retrieve guarantee to retrieve bundle to target_dir

Reply:
Hmmmm. It could be - they should hypothetically be the same. Ill update this.

Reply:
I was a bit confused cause I saw the `return bundle_path` line in the `safe_retrieve`, maybe we just need to change that to `return target_dir ` instead?

Reply:
I removed the return var and just output target_dir instead - seems simple enough. LMK if there is a better way :)